core/Pos/Core/Timestamp.hs

```diff
@@ -59,6 +61,9 @@ diffTimestamp t1 t2 = getTimestamp t1 - getTimestamp t2
 addMicrosecondsToTimestamp :: Microsecond -> Timestamp -> Timestamp
 addMicrosecondsToTimestamp m t = Timestamp { getTimestamp = (getTimestamp t) + m }
 
+timestampToPosix :: Timestamp -> POSIXTime
+timestampToPosix (Timestamp ts) = fromIntegral ts / 1000000
```

Comment:
https://github.com/acw/time-units/blob/master/Data/Time/Units.hs

Or you can "extend" the typeclass by creating a custom one - `TimeUnitCustom` or something and add that function to it and place it locally in `cardano-sl` module.